PROJECT_GUIDELINES.md

```diff
@@ -0,0 +1,173 @@
+
+- Add a semantic [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/).
+- Edit the PR title by adding a semantic prefix like `fix:` or `feat:` or any other [conventional commit type](https://github.com/commitizen/conventional-commit-types/blob/master/index.json).
+
+#### Contributing
+
+Please see our [contributing guidelines](https://github.com/Esri/calcite-app-components/blob/master/CONTRIBUTING.md).
+
+## Code Base
+
+Our code base is written in typescript.
+
+### A11y
+
+We generally follow the guidelines and standards in these articles:
+
+- [Google accessibility overview](https://developers.google.com/web/fundamentals/accessibility/)
+- [WAI-ARIA Authoring Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)
+
+### Components
+
+Have an idea for a new component? Awesome! Check out our [new component checklist](https://github.com/Esri/calcite-app-components/wiki/New-Component-Checklist) and submit an [issue](https://github.com/Esri/calcite-app-components/issues/new?assignees=&labels=new+component%2C+0+-+new%2C+architecture&template=new-component.md&title=New+Component%3A+).
+
+#### Best Practices
+
+We generally follow these best practices when authoring our components:
+
+- [Google Web Component Best Practices](https://developers.google.com/web/fundamentals/web-components/best-practices)
+- [Custom Element Conformance - W3C Editor's Draft](https://html.spec.whatwg.org/multipage/custom-elements.html)
+
+#### Structure
+
+Please see Stencil's [style guide](https://github.com/ionic-team/stencil/blob/master/STYLE_GUIDE.md) for details on component structure.
+
+#### Styling
+
+Our setup uses the shadow DOM, so most of our styles are encapsulated in our calcite design system. We like it this way because it keeps our components consistent across applications.
+
+#### Utils
+
+##### Unique IDs for components
+
+Many times it is necessary for components to have a `id="something"` attribute for things like `<label>` and various `aria-*` properties. To safely generate a unique id for a component but to also allow a user supplied `id` attribute to work follow the following pattern:
+
+```
+import { guid } from "../../utils/guid";
+
+@Component({
+  tag: "calcite-example",
+  styleUrl: "calcite-example.scss",
+  shadow: true
+})
+export class CalciteExample {
+
+  // ...
+
+  guid: string = `calcite-example-${guid()}`;
+
+  render() {
+    const id = this.el.id || this.guid;
+    return (
+      <Host id={id}></Host>
+    );
+  }
+
+  // ...
+}
+```
+
+This will create a unique id attribute like `id="calcite-example-51af-0941-54ae-22c14d441beb"` which should have a VERY low collision change since guid() generates IDs with `window.crypto.getRandomValues`. If a user supplies an `id` this will respect the users `id`.
+
+##### i18n
+
+Components should require as a few text translations as possible. In general lets users supply text values via slots and attributes. The lets user handle translations with their apps.
```

Comment:
> In general lets users supply text values via slots and attributes. The lets user handle translations with their apps.

Can you revisit the `lets users` usage?

Reply:
oh glob! i didn't do a good job proofreading here